File: maxsimum-product-of-two-elements-in-an-array.js
var maxProduct = function(nums) {
  let max1 = -1;
  let max2 = -1;

  for (let i = 0; i < nums.length; i++) {
      if (nums[i] >= max1) {
          max2 = max1;
          max1 = nums[i];
      } else if (nums[i] > max2) {
          max2 = nums[i];
      }
  }

  return (max1 - 1) * (max2 - 1);
};

File: maxsimum-product-of-two-elements-in-an-array.py
class Solution(object):
    def maxProduct(self, nums):
        nums.sort()
        return (nums[-1] - 1) * (nums[-2] - 1)
    # timesort O(n logn)

#İnsertionSort O(n^2)
"""    class Solution(object):
    def maxProduct(self, nums):
        for i in range(len(nums)):
            x = nums[i]
            j = i - 1
            while j >= 0 and x < nums[j]:
                nums[j+1] = nums[j]
                j = j - 1
            nums[j+1] = x
        return (nums[-1] - 1) * (nums[-2] - 1)
"""